sentry/src/main/java/io/sentry/util/StringUtils.java

```diff
@@ -72,4 +80,50 @@ private StringUtils() {}
     }
     return String.format(Locale.ROOT, "%.1f %cB", bytes / 1000.0, ci.current());
   }
+
+  /**
+   * Calculates the SHA-1 String hash
+   *
+   * @param str the String
+   * @param logger the Logger
+   * @return The hashed String or null otherwise
+   */
+  public static @Nullable String calculateStringHash(
+      final @Nullable String str, final @NotNull ILogger logger) {
+    if (str == null || str.isEmpty()) {
+      return null;
+    }
+
+    try {
+      // getInstance() method is called with algorithm SHA-1
+      final MessageDigest md = MessageDigest.getInstance("SHA-1");
+
+      // digest() method is called
+      // to calculate message digest of the input string
+      // returned as array of byte
+      final byte[] messageDigest = md.digest(str.getBytes(UTF_8));
+
+      // Convert byte array into signum representation
+      final BigInteger no = new BigInteger(1, messageDigest);
+
+      // Convert message digest into hex value
+      final StringBuilder stringBuilder = new StringBuilder(no.toString(16));
+
+      // Add preceding 0s to make it 32 bit
+      while (stringBuilder.length() < 32) {
```

Comment:
Do we need a 32 character long string?
I don't believe we're reducing the risk of collision with this so maybe we can go without this altogether. (seems like this came [from here](https://www.geeksforgeeks.org/sha-512-hash-in-java/), they likely needed a fixed size, but we don't).

Reply:
Yep, no need to be fixed size, but iOS also uses something similar https://github.com/getsentry/sentry-cocoa/blob/e254b733f7706d4c04c02093f5ad0715ced92443/Sources/Sentry/SentryDsn.m
`CC_SHA1_DIGEST_LENGTH * 2` so it might be fixed size as well?

Reply:
In iOS is just looks like it's allocating a string twice the size the number of bytes used in the digest creation.
It's a copy paste from:https://stackoverflow.com/questions/7570377/creating-sha1-hash-from-nsstring

See: 
<img width="669" alt="image" src="https://user-images.githubusercontent.com/1633368/168888328-e1f7242c-bcaa-4d1e-911c-04072502629c.png">

Reply:
Updated